app/models/world_location.rb

```diff
@@ -52,28 +53,28 @@ def self.cache_fetch(key)
     end
   end
 
-  def initialize(data)
-    @data = data
-  end
-
-  def slug
-    self.details["slug"]
+  def initialize(location)
+    @location = location
   end
 
   def ==(other)
-    other.is_a?(self.class) && other.slug == slug
+    other.is_a?(self.class) && other.slug == self.slug
```

Comment:
The intent was to mirror the way it was prior to the initial change and somewhat be a bit explicit